src/functions.js

```diff
@@ -121,7 +120,27 @@ var _ = Mavo.Functions = {
 			return n && n > 1? [arr] : arr;
 		}
 
-		return n && n > 1? arr.slice(0, n) : arr[0];
+		if (n && n > 1) {
+			var ret = [];
+
+			for (i = 0; i<arr.length && ret.length<n; i++) {
+
+				if (Mavo.value(arr[i]) !== null) {
+					ret.push(arr[i]);
+				}
+
```

Comment:
Nitpick: I generally agree that blank lines before/after control flow statements and blocks of code help readability, props for that! However, this here may be a little excessive.
I usually avoid blank lines if the block does not have any statements before/after it within the same scope since I feel that indentation sets it apart sufficiently, but no strong opinions here.